src/client/service/identitiesInfoGet.ts

```diff
@@ -18,40 +22,90 @@ function identitiesInfoGet({
   authenticate: Authenticate;
 }) {
   return async (
-    call: grpc.ServerUnaryCall<identitiesPB.Provider, identitiesPB.Provider>,
-    callback: grpc.sendUnaryData<identitiesPB.Provider>,
+    call: grpc.ServerWritableStream<
+      identitiesPB.ProviderSearch,
+      identitiesPB.Info
+    >,
   ): Promise<void> => {
+    const genWritable = grpcUtils.generatorWritable(call);
     try {
-      const response = new identitiesPB.Provider();
       const metadata = await authenticate(call.metadata);
       call.sendMetadata(metadata);
       const {
-        providerId,
+        providerIds,
+        identityId,
       }: {
-        providerId: ProviderId;
+        providerIds: Array<ProviderId>;
+        identityId: IdentityId;
       } = validateSync(
         (keyPath, value) => {
           return matchSync(keyPath)(
-            [['providerId'], () => validationUtils.parseProviderId(value)],
+            [['providerIds'], () => value.map(validationUtils.parseProviderId)],
+            [['identityId'], () => validationUtils.parseIdentityId(value)],
             () => value,
           );
         },
         {
-          providerId: call.request.getProviderId(),
+          providerIds: call.request.getProviderIdList(),
+          identityId: call.request.getIdentityId(),
         },
       );
-      const provider = identitiesManager.getProvider(providerId);
-      if (provider !== undefined) {
-        const identities = await provider.getAuthIdentityIds();
-        response.setProviderId(providerId);
-        if (identities.length !== 0) {
-          response.setIdentityId(identities[0]);
+      // Process options that were set
+      if (providerIds.length === 0) {
+        Object.keys(identitiesManager.getProviders()).forEach((id) =>
+          providerIds.push(id as ProviderId),
+        );
+      }
+      const searchTerms = call.request.getSearchTermList();
+      const getDisconnected = call.request.getDisconnected();
+      if (getDisconnected) {
+        // Currently this command performs the same way regardless of whether
+        // this option is set (i.e. always disconnected)
+      }
+      const identities: Array<IdentityData | undefined> = [];
+      for (const providerId of providerIds) {
+        // Get provider from id
+        const provider = identitiesManager.getProvider(providerId);
+        if (provider === undefined)
+          throw new identitiesErrors.ErrorProviderMissing();
+        // Get our own authenticated identity in order to query, skip provider
+        // if not authenticated
+        const authIdentities = await provider.getAuthIdentityIds();
+        if (authIdentities.length === 0) {
+          break;
+        }
+        // Get identity data
+        identities.push(
+          await provider.getIdentityData(authIdentities[0], identityId),
+        );
```

Comment:
Another case where we only query one of the authenticated identities of the provider - this seems like a potentially worse case, where we only attempt to get an identity from one of our own authenticated identities. My assumption is that we can only get identity data of `identityId` if it's connected to the `authIdentity` passed?

Reply:
It's not really an issue for `identitiesInfoGet` since you're just querying the provider to get information about an identity. You don't need to be connected to it, you just need any authenticated identity in order to perform the request.

Reply:
Great, wasn't too sure what info we were getting from the request, and whether it meant we'd need to be connected to it.